react-app/src/components/projects/ProjectDialog.tsx

```diff
@@ -489,3 +491,66 @@ export const ProjectAgencyResponseDialog = (props: IProjectAgencyResponseDialog)
     </ConfirmDialog>
   );
 };
+
+interface INotificationDialog {
+  initialValues: ProjectGet;
+  open: boolean;
+  ungroupedAgencies: Partial<Agency>[];
+  postSubmit: () => void;
+  onCancel: () => void;
+}
+
+export const ProjectNotificationDialog = (props: INotificationDialog) => {
+  const api = usePimsApi();
+  const { initialValues, open, postSubmit, onCancel } = props;
+  const [rows, setRows] = useState([]);
+  const [submitting, setSubmitting] = useState(false);
+  useEffect(() => {
+    if (initialValues) {
+      setRows(
+        initialValues.Notifications?.map((resp) => ({
+          id: resp.ChesMessageId,
+          agency: props.ungroupedAgencies?.find((agc) => agc.Id === resp.ToAgencyId)?.Name,
+          to: resp.To,
+          subject: resp.Subject,
+          status: getStatusString(resp.Status),
+          sendOn: resp.SendOn,
+          projectNumber: props.initialValues.ProjectNumber,
+        })),
+      );
+    }
+  }, [initialValues]);
+
+  const fetchUpdatedNotifications = async (projectId: number) => {
+    setSubmitting(true);
+    try {
+      const notifications = await api.notifications.getNotificationsByProjectId(projectId);
+      setRows(notifications);
+    } catch (error) {
+      console.error('Error fetching updated notifications:', error);
+    } finally {
+      setSubmitting(false);
```

Comment:
This is fine but alternatively you could leverage the useDataSubmitter hook which handles the boilerplate of exposing some loading state for the request.